jax/api.py

```diff
@@ -857,11 +864,25 @@ def pmap(fun, axis_name=None, devices=None, backend=None):
   """
   _check_callable(fun)
   axis_name = _TempAxisName(fun) if axis_name is None else axis_name
+  if isinstance(static_argnums, int):
+    static_argnums = (static_argnums,)
 
   @wraps(fun)
   def f_pmapped(*args, **kwargs):
     f = lu.wrap_init(fun)
-    args, in_tree = tree_flatten((args, kwargs))
+    if static_argnums:
+      # Check no static arguments have been shareded
+      for i in static_argnums:
+        if type(args[i]) is pxla.ShardedDeviceArray:
+          raise ValueError("The %d argument is a ShardedDeviceArray, but it has"
```

Comment:
nit: "Argument %d" instead of "The %d argument", I think something like "The 4 argument" is confusing.